internal/pkg/cachecli/cache_clean.go

```diff
@@ -0,0 +1,221 @@
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+)
+
+// CleanLibraryCache : clean only library cache, will return a error if one occurs
+func CleanLibraryCache() error {
+	sylog.Debugf("Removing: %v", cache.Library())
+
+	err := os.RemoveAll(cache.Library())
+	if err != nil {
+		sylog.Warningf("Unable to clean cache: %v", err)
+	}
+
+	return err
+}
+
+// CleanOciCache : clean only oci cache, will return a error if one occurs
+func CleanOciCache() error {
+	sylog.Debugf("Removing: %v", cache.OciTemp())
+
+	err := os.RemoveAll(cache.OciTemp())
+	if err != nil {
+		sylog.Warningf("Unable to clean cache: %v", err)
+	}
+
+	return err
+}
+
+// CleanBlobCache : clean only blob cache, will return a error if one occurs
+func CleanBlobCache() error {
+	sylog.Debugf("Removing: %v", cache.OciBlob())
+
+	err := os.RemoveAll(cache.OciBlob())
+	if err != nil {
+		sylog.Warningf("Unable to clean cache: %v", err)
+	}
+
+	return err
+
+}
+
+func cleanLibraryCache(cacheName string) (bool, error) {
+	foundMatch := false
+	libraryCacheFiles, err := ioutil.ReadDir(cache.Library())
+	if err != nil {
+		sylog.Warningf("Unable to opening cache folder: %v", err)
+		return false, err
+	}
+	for _, f := range libraryCacheFiles {
+		cont, err := ioutil.ReadDir(filepath.Join(cache.Library(), f.Name()))
+		if err != nil {
+			sylog.Warningf("Unable to look in cache folder: %v", err)
+			return false, err
+		}
+		for _, c := range cont {
+			if c.Name() == cacheName {
+				sylog.Debugf("Removing: %v", filepath.Join(cache.Library(), f.Name(), c.Name()))
+				err = os.RemoveAll(filepath.Join(cache.Library(), f.Name(), c.Name()))
+				if err != nil {
+					sylog.Warningf("Unable to remove cache: %v", err)
+					return false, err
+				}
+				foundMatch = true
+			}
+		}
+	}
+
+	return foundMatch, nil
+}
+
+func cleanOciCache(cacheName string) (bool, error) {
+	foundMatch := false
+	blobs, err := ioutil.ReadDir(cache.OciTemp())
+	if err != nil {
+		sylog.Warningf("Unable to opening cache folder: %v", err)
+		return false, err
+	}
+	for _, f := range blobs {
+		blob, err := ioutil.ReadDir(filepath.Join(cache.OciTemp(), f.Name()))
+		if err != nil {
+			sylog.Warningf("Unable to look in cache folder: %v", err)
+			return false, err
+		}
+		for _, b := range blob {
+			if b.Name() == cacheName {
+				sylog.Debugf("Removing: %v", filepath.Join(cache.OciTemp(), f.Name(), b.Name()))
+				err = os.RemoveAll(filepath.Join(cache.OciTemp(), f.Name(), b.Name()))
+				if err != nil {
+					sylog.Warningf("Unable to remove cache: %v", err)
+					return false, err
+				}
+				foundMatch = true
+			}
+		}
+	}
+
+	return foundMatch, nil
+}
+
+// CleanCacheName : clean a cache with a specific name (cacheName). if libraryCache == true; search only the
+// library. if ociCache == true; search the oci cache. returns false if no match found. Will return error if any occurs
+func CleanCacheName(cacheName string, libraryCache, ociCache bool) (bool, error) {
```

Comment:
So, I think the name of this function is maybe slightly unclear. It should be clarified that this `cacheName` is actually the name of an image within the cache, not the name of which cache to go through.